.github/workflows/publish-to-galaxy.yml

```diff
@@ -1,32 +1,88 @@
+        type: choice
+        options:
+          - Release
+          - Dry Run
 
 permissions:
   contents: read
 
 jobs:
+  validate:
+    name: Setup
+    runs-on: ubuntu-22.04
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Branch check
+        if: ${{ github.event.inputs.release_type != 'Dry Run' }}
+        run: |
+          if [[ "$GITHUB_REF" != "refs/heads/rc" ]] && [[ "$GITHUB_REF" != "refs/heads/hotfix-rc" ]]; then
+            echo "==================================="
+            echo "[!] Can only release from the 'rc' or 'hotfix-rc' branches"
+            echo "==================================="
+            exit 1
+          fi
+
+      - name: Get version
+        id: version
+        run: |
+          VERSION=$(cat galaxy.yml | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+")
+          echo "version=$VERSION"
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
   publish:
     runs-on: ubuntu-latest
+    needs: validate
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - name: Retrieve secrets from Secrets Manager
-        uses: bitwarden/sm-action@92d1d6a4f26a89a8191c83ab531a53544578f182 # v2.0.0
+      - name: Login to Azure - CI Subscription
+        uses: Azure/login@e15b166166a8746d1a47596803bd8c1b595455cf # v1.6.0
         with:
-          access_token: ${{ secrets.BWS_ACCESS_TOKEN }}
-          secrets: |
-            dc3dc07b-1bbe-4ca6-bb16-b0d500fd1b71 > GALAXY_API_KEY
+          creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
 
-      - name: Publish to Ansible Galaxy
-        uses: ansible/ansible-publish-action@a56a0328c92c1d4feedf5bd7f7cf7ec7a4ae3f09 # v1.0.0
+      - name: Retrieve github PAT secrets
+        id: retrieve-secret-pat
+        uses: bitwarden/gh-actions/get-keyvault-secrets@main
         with:
-          api_key: ${{ env.GALAXY_API_KEY }}
-          src_path: .
+          keyvault: "bitwarden-ci"
+          secrets: "ansible-galaxy-api-key"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade ansible
+          pip install -r requirements.txt
+          pip install --no-index --find-links=dist bitwarden_sdk
+
+      - name: Build Ansible Collection
+        run: ansible-galaxy collection build . --force
+
+      - name: Create release
+        if: ${{ github.event.inputs.release_type != 'Dry Run' }}
+        uses: ncipollo/release-action@6c75be85e571768fa31b40abf38de58ba0397db5 # v1.13.0
+        with:
+          tag: v${{ needs.validate.outputs.version }}
+          name: v${{ needs.validate.outputs.version }}
+          body: "<insert release notes here>"
+          draft: true
+          artifact: bitwarden-secrets-${{ needs.validate.outputs.version }}.tar.gz
+
+      - name: Publish to Ansible Galaxy
```

Comment:
Just curious: is there any reason for not using the prebuilt Action for publishing?

Reply:
Since it's one liner I used it instead of adding dependency for a new action. Do you think it's better to use ansible action @tangowithfoxtrot ?

Reply:
I think it's fine to use the one-liner. I was just curious :)